### DISCUSSION TITLE
How to stop execution until the first terminal command succeeds  ?

### DISCUSSION THREAD

**Discussion by jaybhanushali3195** [ttscoff/bunch-docs#293] [Q&A]
Hello,
I am trying to establish a workflow where the following needs to happen in sequence
1) Open Iterm, enter authentication command (This I have already done)
2) I manually press a yubikey to authenticate
3) If is successfull, launch other iterm and write some commands

The problem I am facing is that bunch is not waiting for me to complete the authentication.
Any way to solve this ?

**Comment by ttscoff** [comment]
Have you tried setting the execution order to sequential in the front 
matter?

```
---
sequence: sequential
---
```

That may or may not help. You can also play around with using the `run 
before` frontmatter. That should execute in its entirety before 
executing the rest of the Bunch, and you could have the script return a 
frontmatter key/value that is used in an if/then statement in the Bunch 
to determine if the auth was successful.

It's obviously a bit of a tricky situation, and one that Bunch isn't 
well-designed to handle.

-Brett

**Reply by jaybhanushali3195** [reply]
Tried it no luck :( actually delay is also not working , I tried delay with various options eg. ( pause 10) , delay ( 10 ) , ~ 10 , none of them works. I will make a different post for it.
I tried run before: auth.sh and it also seems to not trigger any script before launching bunch
